bodhi/server/buildsys.py

```diff
@@ -314,18 +315,20 @@ def get_krb_conf(config):
 
 def get_session():
     """ Get a new buildsystem instance """
-    global _buildsystem
-    if not _buildsystem:
-        log.warning('No buildsystem configured; assuming testing')
-        return DevBuildsys()
-    return _buildsystem()
+    global _buildsystem, _buildsystem_login_lock
+    with _buildsystem_login_lock:
+        if not _buildsystem:
+            log.warning('No buildsystem configured; assuming testing')
+            return DevBuildsys()
+        return _buildsystem()
 
 
 def setup_buildsystem(settings):
-    global _buildsystem, _koji_hub
+    global _buildsystem, _koji_hub, _buildsystem_login_lock
     if _buildsystem:
         return
 
+    _buildsystem_login_lock = Lock()
```

Comment:
Okay, cool. I'm inclined to change it to initialize at the module level since that's less surprising to me.